src/Codec/DecodeIfSupported.php

```diff
@@ -25,27 +25,22 @@
  */
 trait DecodeIfSupported
 {
-    /**
-     * @param mixed $value
-     * @psalm-assert-if-true BSONType $value
-     */
-    abstract public function canDecode($value): bool;
+    /** @psalm-assert-if-true BSONType $value */
+    abstract public function canDecode(mixed $value): bool;
 
     /**
-     * @param mixed $value
      * @psalm-param BSONType $value
      * @return mixed
      * @psalm-return NativeType
      * @throws UnsupportedValueException if the decoder does not support the value
      */
-    abstract public function decode($value);
+    abstract public function decode(mixed $value);
```

Comment:
Confirmed this is *not* a breaking change for classes that override this abstract method. https://3v4l.org/donQG

I don't get why Symfony BC promise prohibit adding typehint to arguments in classes. https://symfony.com/doc/current/contributing/code/bc.html#changing-classes

Reply:
Correct. Without a type specified for an argument, people can't add their own, narrower types. When we add a type, people are still allowed to widen types or omit the type entirely, which is why we can add parameter types in classes and interfaces in minor versions.